src/Adapter/Contact/QueryHandler/GetContactForEditingHandler.php

```
@@ -0,0 +1,98 @@
+use PrestaShop\PrestaShop\Core\Domain\Contact\QueryHandler\GetContactForEditingHandlerInterface;
+use PrestaShopException;
+use Symfony\Component\Form\DataTransformerInterface;
+
+/**
+ * Class GetContactForEditingHandler is responsible for getting the data for contact edit page.
+ *
+ * @internal
+ */
+final class GetContactForEditingHandler implements GetContactForEditingHandlerInterface
+{
+    /**
+     * @var DataTransformerInterface
+     */
+    private $stringArrayToIntegerArrayDataTransformer;
+
+    /**
+     * @param DataTransformerInterface $stringArrayToIntegerArrayDataTransformer
+     */
+    public function __construct(DataTransformerInterface $stringArrayToIntegerArrayDataTransformer)
+    {
+        $this->stringArrayToIntegerArrayDataTransformer = $stringArrayToIntegerArrayDataTransformer;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws ContactException
+     */
+    public function handle(GetContactForEditing $query)
+    {
+        try {
+            $contact = new Contact($query->getContactId()->getValue());
+
+            if (0 >= $contact->id) {
+                throw new ContactNotFoundException(
+                    sprintf(
+                        'Contact object with id %s was not found',
+                        var_export($query->getContactId()->getValue(), true)
+                    )
+                );
+            }
+
+            $editableContact = new EditableContact(
+                $query->getContactId()->getValue(),
+                $contact->name,
+                $contact->email,
+                $contact->customer_service,
+                $contact->description,
+                $this->stringArrayToIntegerArrayDataTransformer->reverseTransform($contact->getAssociatedShops())
+            );
+        } catch (PrestaShopException $e) {
+            throw new ContactException(
+                sprintf(
+                    'An unexpected error occurred when retrieving contact with id %s',
```

Is it a wording here?

---

No - these kind of errors which exist in the Exception scope are not translatable - usually these kind of errors are being later handled with the translatable message which is more friendly for regular user. If the error is not handled then it throws an error with the message in exception if the debug mode is on.

---

@tomas862, thanks for the reply!